homeassistant/components/light/qwikswitch.py

```diff
@@ -4,18 +4,34 @@
 For more details about this platform, please refer to the documentation at
 https://home-assistant.io/components/light.qwikswitch/
 """
-import logging
+from homeassistant.components.qwikswitch import (
+    QSToggleEntity, DOMAIN as QWIKSWITCH)
+from homeassistant.components.light import SUPPORT_BRIGHTNESS, Light
 
-DEPENDENCIES = ['qwikswitch']
+DEPENDENCIES = [QWIKSWITCH]
 
 
-# pylint: disable=unused-argument
-def setup_platform(hass, config, add_devices, discovery_info=None):
+async def async_setup_platform(hass, _, add_devices, discovery_info=None):
     """Add lights from the main Qwikswitch component."""
-    if discovery_info is None:
-        logging.getLogger(__name__).error(
-            "Configure Qwikswitch Light component failed")
-        return False
+    qsusb = hass.data[QWIKSWITCH]
+    devs = [QSLight(id, qsusb) for id in discovery_info[QWIKSWITCH]]
 
-    add_devices(hass.data['qwikswitch']['light'])
-    return True
+    add_devices(devs)
+
+    for _id, dev in zip(discovery_info[QWIKSWITCH], devs):
+        hass.helpers.dispatcher.async_dispatcher_connect(
```

Comment:
Move this to the entity coroutine method `async_added_to_hass`, ie connect each entity from within itself. Reason is that `add_devices` schedules a call to add devices, so `add_devices` is not guaranteed to have added the entities when it returns.